### PR TITLE
Fix/action btn loading in table widget

### PR DESCRIPTION
**Pull request by vicky-primathon**
Action button in table widget made to go in loading state during click operation is completed. Code refactored, ReactTableComponent is made stateless. Use useMemo to stop unwanted re-rendering of table cells.

### CODE REVIEW

File: app/client/src/components/designSystems/appsmith/Table.tsx [hunk 1]
```diff
@@ -62,12 +55,9 @@ export const Table = (props: TableProps) => {
     }),
     [],
   );
-
-  const pageCount = Math.ceil(data.length / props.pageSize);
+  const pageCount = Math.ceil(props.data.length / props.pageSize);
   const currentPageIndex = props.pageNo < pageCount ? props.pageNo : 0;
-  // const filteredColumns = columns.filter((column: ReactTableColumnProps) => {
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/Table.tsx:68]: We missed a commented block of code.

File: app/client/src/components/designSystems/appsmith/Table.tsx [hunk 2]
```diff
@@ -128,73 +118,27 @@ export const Table = (props: TableProps) => {
...
                 {headerGroup.headers.map((column: any, columnIndex: number) => {
-                  if (column.isResizing) {
-                    props.handleResizeColumn(
-                      columnIndex,
-                      column.getHeaderProps().style.width,
-                    );
-                  }
                   return (
-                    <div
+                    <RenderColumnHeader
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/Table.tsx:127]: We don't need "Render" in the name of a component. Can we use a name which identifies exactly what this component is?

File: app/client/src/components/designSystems/appsmith/TableColumnMenu.tsx [hunk 3]
```diff
@@ -31,22 +32,17 @@ export const TableColumnMenuPopup = (props: TableColumnMenuPopup) => {
...
     >
-      <Icon
-        icon="more"
-        iconSize={12}
-        color="#A1ACB3"
-        onClick={e => {
-          props.showMenu(props.columnIndex);
-        }}
-      />
+      <Icon icon="more" iconSize={12} color="#A1ACB3" />
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/TableColumnMenu.tsx:35]: Let us use the "more_vertical_icon" from out controlicons.

Comment (vicky-primathon) [reply app/client/src/components/designSystems/appsmith/TableColumnMenu.tsx:35]: @riodeuno leaving it as such. Will remove it once we have these column action handled from pane properties

**Pull request merged**